src/main/java/uk/gov/pay/ledger/payout/service/PayoutService.java

```diff
@@ -21,4 +31,31 @@ public void upsertPayoutFor(EventDigest eventDigest) {
         PayoutEntity payoutEntity = payoutEntityFactory.create(eventDigest);
         payoutDao.upsert(payoutEntity);
     }
+
+    public PayoutSearchResponse searchPayouts(PayoutSearchParams searchParams, UriInfo uriInfo) {
+        List<PayoutView> payoutViewList = payoutDao.searchPayouts(searchParams)
+                .stream()
+                .map(PayoutView::from)
+                .collect(Collectors.toList());
+
+        Long total = payoutDao.getTotalForSearch(searchParams);
+
+        long size = searchParams.getDisplaySize();
+        if (total > 0 && searchParams.getDisplaySize() > 0) {
+            long lastPage = (total + size - 1) / size;
+            if (searchParams.getPageNumber() > lastPage || searchParams.getPageNumber() < 1) {
+                throw new WebApplicationException("the requested page not found",
```

Comment:
This matches parity from the current method of doing things which is good but potentially 
```
"The requested page was not found"
```